src/Models/HydrostaticFreeSurfaceModels/update_hydrostatic_free_surface_model_state.jl

```diff
@@ -28,8 +28,380 @@ they are called in the end.
 update_state!(model::HydrostaticFreeSurfaceModel, callbacks=[]; compute_tendencies = true) =
          update_state!(model, model.grid, callbacks; compute_tendencies)
 
-function update_state!(model::HydrostaticFreeSurfaceModel, grid, callbacks; compute_tendencies = true)
+function fill_cubed_sphere_halo_regions!(field, ::Tuple{<:Center, <:Center})
```

Comment:
@siddharthabishnu why there is a second positional argument here if it's not used???
just to differentiate between methods?

If that's why and if these locations are the corresponding locations of the fields, then could you please dispatch on the field parameters? it's much cleaner and how we generally do things

Reply:
Also, please:

- move these methods to somewhere relevant, potentially within MultiRegion module as we discussed.
- use the `CubedSphereField` type,
  https://github.com/CliMA/Oceananigans.jl/blob/fb9455bc10721a880d9911a39735e37f1c4961e0/src/MultiRegion/cubed_sphere_field.jl#L5
    to dispatch on these methods to ensure they don't do anything to Fields on grids other than cubed-sphere grids.

Reply:
@navidcy, I will make these adjustments today.

Reply:
the location of the field is a parameter of the Field type; you can infer it from there

Reply:
It's done, @navidcy. Check out the last commit [1bf9119](https://github.com/CliMA/Oceananigans.jl/pull/3488/commits/1bf911929ccd7853ca868e24c2f76dded937b141).